src/EnergyPlus/General.cc

```diff
@@ -733,14 +707,11 @@ bool BetweenDates(int const TestDate,  // Date to test
     // REFERENCES:
     // Adapted from BLAST BTWEEN function.
 
-    // Return value
-    bool BetweenDates;
-
-    BetweenDates = false; // Default case
+    bool BetweenDates = false; // Default case
 
     if (StartDate <= EndDate) { // Start Date <= End Date
         if (TestDate >= StartDate && TestDate <= EndDate) BetweenDates = true;
-    } else { // EndDate <= StartDate
+    } else { // EndDate < StartDate
         if (TestDate <= EndDate || TestDate >= StartDate) BetweenDates = true;
     }
```

Comment:
Feels like there should be an additional simplification here, but this is totally fine.